src/main/java/org/opendevstack/provision/services/JenkinsPipelineAdapter.java

```diff
@@ -235,73 +230,86 @@ public OpenProjectData createPlatformProjects(OpenProjectData project) throws IO
-              httpPreAuthenticatedCall(HttpVerb.GET)
-                  .url(jobsUrl)
-                  .queryParam("groupPath", group)
-                  .returnTypeReference(new TypeReference<List<Job>>() {}));
-      if (jobs == null) {
-        jobs = new ArrayList<>();
-      }
-      List<Job> enabledJobs = jobs.stream().filter(Job::isEnabled).collect(Collectors.toList());
-      return enabledJobs;
-    } catch (IOException ex) {
-      logger.error(GENERIC_RUNDECK_ERRMSG, ex);
-    }
-    return new ArrayList<>();
-  }
-
   @Override
   public Map<String, String> getProjects(String filter) {
     throw new NotImplementedException();
   }
 
   @Override
   public String getAdapterApiUri() {
-    return rundeckApiUrl;
+    throw new NotImplementedException();
   }
 
-  private ExecutionsData prepareAndExecuteJob(String jobNameOrId, Map<String, String> options)
+  private ExecutionsData prepareAndExecuteJob(
+      String jobNameOrId, Map<String, String> options, String webhook_proxy_secret)
       throws IOException {
     Preconditions.checkNotNull(jobNameOrId, "Cannot execute Null Job!");
 
-    Job job = jobStore.getJobByNameOrId(jobNameOrId);
+    String url;
+    Execution execution = new Execution();
 
-    String url = String.format("%s%s/job/%s/run", rundeckUri, rundeckApiPath, job.getId());
+    String projID = Objects.toString(options.get("PROJECT_ID"));
 
-    Execution execution = new Execution();
+    if (projectCreateOpenshiftJob.equals(jobNameOrId)) { // we create something
+
+      Job job = getCreateOdsProjectsJob();
+
+      String webhookProxyHost =
+          String.format(
+              projectOpenshiftJenkinsWebhookProxyNamePattern, "prov", projectOpenshiftBaseDomain);
+      url =
+          "https://"
+              + webhookProxyHost
+              + "/build?trigger_secret="
+              + webhook_proxy_secret
+              + "&jenkinsfile_path="
+              + job.jenkinsfilePath
+              + "&component=ods-corejob-"
+              + job.name
+              + "-"
+              + projID;
+      execution.branch = job.branch;
+      execution.repository = job.gitURL;
+
+    } else {
+      String component_id = Objects.toString(options.get("component_id"));
+
+      final Job job =
+          getQuickstarters().stream().filter(x -> x.id.equals(jobNameOrId)).findFirst().get();
+
+      String webhookProxyHost =
+          String.format(
+              projectOpenshiftJenkinsWebhookProxyNamePattern, projID, projectOpenshiftBaseDomain);
+      url =
+          "https://"
+              + webhookProxyHost
+              + "/build?trigger_secret="
+              + webhook_proxy_secret
+              + "&jenkinsfile_path="
+              + job.jenkinsfilePath
+              + "&component=ods-quickstarter-"
+              + job.getName()
+              + "-"
+              + component_id;
+      execution.branch = job.branch;
+      execution.repository = job.gitURL;
+    }
     if (options != null) {
```

Comment:
Took me a long time to realise that HERE all the magic happens (e.g. setting the project to create). Also, it seems weird that we already use `options` above, assuming it is non-null and then have this check here ...

Can we make the flow clearer?